### PR TITLE
Update Sugar client token handling

### PR DESCRIPTION
**Pull request by eatyourgreens**
`auth._bearerToken` is undefined as of 3.3.0, so pass the token to Sugar from `auth.checkBearerToken()`.
Check the current token before each Sugar API request.

### CODE REVIEW

File: lib/sugar.js [hunk 1]
```diff
@@ -10,16 +10,23 @@ if (typeof navigator !== 'undefined') {
   var sugarApiClient = new JSONAPIClient(config.sugarHost, {
     'Content-Type': 'application/json',
     'Accept': 'application/json',
+  }, {
+    beforeEveryRequest: function() {
+      auth.checkBearerToken()
+        .then(function () {
+          return null;
+        });
+    }
```
Comment (wgranger) [inline lib/sugar.js:19]: This mixin inclusion crashes PFE when accessing a project home page.

File: lib/sugar.js [hunk 2]
```diff
@@ -10,16 +10,23 @@ if (typeof navigator !== 'undefined') {
   var sugarApiClient = new JSONAPIClient(config.sugarHost, {
     'Content-Type': 'application/json',
     'Accept': 'application/json',
+  }, {
+    beforeEveryRequest: function() {
+      auth.checkBearerToken()
```
Comment (eatyourgreens) [inline lib/sugar.js:15]:
```suggestion
      return auth.checkBearerToken()
```

File: lib/sugar.js (hunk 1 above)
Comment (eatyourgreens) [reply lib/sugar.js]: Good catch. I missed out a `return`.

**Comment by eatyourgreens** [comment]
Testing this, I noticed that PFE is still on `3.0.0-rc.0` so is behind the latest client by a couple of minor versions.

**Pull request merged**